sys/kern/kern_module.c

```diff
@@ -415,12 +416,17 @@ kern_modstat(struct thread *td, int modid,
 	    version == sizeof(struct module_stat_v2));
 	if (!is_v1v2 && version != sizeof(struct module_stat))
 		return (EINVAL);
+	if (is_v1v2)
+		user_namep =
+		    &((struct module_stat_v1 * __capability)stat)->name[0];
+	else
+		user_namep = &stat->name[0];
```

Comment:
Can we use plain array decay here instead of &X[0]?

Reply:
Should hopefully do the right thing in hybrid mode now?

Reply:
Upstream uses the &x[0] pattern which is why I kept it.  Would it be better to fix that upstream as well?

Reply:
That doesn't really matter unless we compile with very-aggressive subobject bounds (which is extremely unlikely to work). Personally I prefer array decay, but don't really mind either way.